### DISCUSSION TITLE
Exporting a NN

### DISCUSSION THREAD

**Discussion by NicolasFranklin** [chulminy/CIVE497-CIVE700#141] [General]
Out of curiosity, how we would actually use our trained Neural Networks in practice? Is there a way to export the weights and biases of a trained NN for use with a new unknown dataset? What would this code look like? I know this is outside of the scope of the course, but I am just curious.

Also, are there any more resources we can look towards to extend on our Matlab-based machine learning knowledge?

Thanks!

**Comment by chulminy** [comment]
Yes. We know how we design the architecture of NN. The training process is just to find the parameters (W and b) in NN. 
You can consider it as a "function" but cann't represent it as a simple form. 

Yes. They are just values. You can store them in any format. 

Since we are familiar with functions that can be expressed by math equations, it's not easy to imagine how to store them. 
However, NN is a set of operations using w and b to compute the value from the input.

Rather than looking at the new material, you can look into the code that I provided. 
That's a good starting point. 

There are many NN tutorial online but they mainly talk about how to use NN library, not build it from scratch.
